src/main/java/com/amazonaws/encryptionsdk/internal/DecryptionHandler.java

```diff
@@ -84,11 +88,31 @@ private DecryptionHandler(final CryptoMaterialsManager materialsManager) {
         Utils.assertNonNull(materialsManager, "materialsManager");
 
         this.materialsManager_ = materialsManager;
+        maxBodySize_ = null;
+        maxHeaderSize_ = null;
+        ciphertextHeaders_ = new CiphertextHeaders();
+        ciphertextFooters_ = new CiphertextFooters();
+    }
+
+    private DecryptionHandler(final CryptoMaterialsManager materialsManager,
+            final Integer maxBodySize,
+            final Integer maxHeaderSize) {
+        Utils.assertNonNull(materialsManager, "materialsManager");
+        if (maxBodySize != null && maxBodySize < 0) {
+            throw new IllegalArgumentException("maxBodySize must be null or non-negative.");
+        } else if (maxHeaderSize != null && maxHeaderSize <= 0) {
+            throw new IllegalArgumentException("maxHeaderSize must be null or positive.");
```

Comment:
Here, we specifically make this `<= 0` so I assume the earlier ones should be updated accordingly.